Update documentation to v2.0.0 (#50)
eProsima/Micro-XRCE-DDS-docs: docs/client.rst

    *   - :code:`UCLIENT_SERIAL_TRANSPORT_MTU`
        - This value corresponds to the *Maximum Transmission Unit (MTU)* that can |br|
          be sent and/or received by Serial. It is measured in bytes and, internally, it |br|
          corresponds to the creation of a buffer this size.
        - :code:`<number>`
        - :code:`512`

.. _read_access:

Read Access Delivery Control
----------------------------

The Read Access Delivery Control handles the read operation from a *datareader* previously created
on the *Agent* to fetch data from the middleware.
It comes with an optional ``control`` argument, that allows the *Client* setting the following parameters:

* ``max_bytes_per_second``: Maximum rate at which data messages may be returned, measured in bytes per secpond.
* ``max_elapsed_time``: Maximum amount of time that can be spent by the Agent in delivering the topic, measured in seconds.
* ``max_samples``: Maximum number of topics that the Agent can send to the Client.
* ``min_pace_period``: Minimum elapsed time between two topics deliveries, measured in milliseconds,.

For more information, consult the :ref:`read_access_api` of the :ref:`client_api_label`.

> Creation Mode

Reviewer: For me, it is very confusing that this section is called "Creation mode" and the one right after is called "Creation Mode table", yet they verse about very different things. We should come up with a clearer way to re-structure this

-------------

> The creation of :ref:`entities_label>` on the *Agent* which can act on behalf of the *Clients*

Reviewer: ```suggestion
The creation of :ref:`entities_label` on the *Agent* which can act on behalf of the *Clients*
```

        
        <!-- TOPIC -->
        const char* topic_xml = "<dds>"
                                    "<topic>"
                                        "<name>[TOPIC NAME]</name>"
                                        "<dataType>[TOPIC TYPE]</dataType>"
                                    "</topic>"
                                "</dds>"

    As detailed in the :ref:`getting_started_label` section, *participants*, *topics*, *datawriters*, *datareaders*, *requesters* and *repliers* work similarly.
    *Publishers* and *subscribers*, instead, inherit their XML fields from their associated *dataWriters* and *dataReaders*.
    
    Creation by XML has the advantage of being configurable direclty within the *Client* application,
    but comes with the drawback of offering a very limited set of options as regards the QoS with which the DDS entities
    profiles can be configured. Indeed, only best-effort or reliable communication streams can be set with this creation mode.
    In many cases, these QoS configurations alone may not be enough. For these cases, *eProsima Micro XRCE-DDS* allows the users
    to use the creation by references mode.

> References

Reviewer: I think having an example here would be appropriate, or at least a link to this reference file: https://github.com/eProsima/Micro-XRCE-DDS-Agent/blob/master/agent.refs

Find out more in the :ref:`transport_api` section of the :ref:`client_api_label`.

Discovery profile
^^^^^^^^^^^^^^^^^

The discovery profile allows discovering *Agents* in the network by UDP.
The reachable *Agents* will respond to the discovery call sending information about themselves, as their IP and port.
This can happen in two ways: multicast or unicast.
The discovery phase can be performed before the `uxr_create_session` call to determine the *Agent* to connect with.
The declaration of these functions can be found in ``uxr/client/profile/discovery/discovery.h``.
This profile is enabled when the :code:`UCLIENT_DISCOVERY_PROFILE` is :code:`ON`.

Find out more in the :ref:`dedicated section <discovery_api>` of the API.

.. note::
    This feature is only available on Linux.

Framing profile
^^^^^^^^^^^^^^^

> .. TODO: add info/description

Reviewer 2: @pablogs9 TODO

Reviewer 2: @pablogs9 This is done, right?

Framing profile
^^^^^^^^^^^^^^^

> Framing profile enables :ref:`HDLC Framing <_stream_framing_label>` for using :ref:`stream-oriented transports <_intro_transport>` such as Serial transports or custom transports that required framing.

Reviewer 2: ```suggestion
The framing profile enables :ref:`HDLC Framing <stream_framing_label>` for using :ref:`stream-oriented transports <intro_transport>` such as Serial transports or Custom transports that require framing.
```

------

.. code-block:: c

    void uxr_create_session_retries(uxrSession* session, size_t retries);

> Creates a new session with the *Agent* with retries.

Reviewer 2: ```suggestion
Attempts to establish a new session on the *Agent* :code:`retries` times
```

Transport
`````````
These functions are platform dependent.

> The declaration of these functions can be found in ``uxr/client/profile/transport/`` folder.

Reviewer 2: ```suggestion
The declaration of these functions can be found in the ``uxr/client/profile/transport/`` folder.
```

> The common init transport functions follow the next nomenclature.

Reviewer 2: ```suggestion
The common init transport functions follow the nomenclature below.
```

:transport: The transport to close.

------

.. code-block:: c

    bool uxr_ping_agent(const uxrCommunication* comm, const int timeout);

[TODO]

:comm: [TODO]
:timeout: [TODO]

------

.. code-block:: c

    bool uxr_ping_agent_attempts(const uxrCommunication* comm, const int timeout, const uint8_t attempts);

> [TODO]

Author: @jamoralp
